tests/Unit/ControlSystem/Systems/Test_Shape.cpp

```diff
@@ -0,0 +1,513 @@
+
+  // Allocate now and reuse as we go
+  auto initial_ah_coefs = zero_array;
+  domain::FunctionsOfTime::PiecewisePolynomial<DerivOrder> expected_ah_coefs{};
+
+  const std::string deriv_order_string =
+      "DerivOrder=" + get_output(DerivOrder) + ": ";
+  {
+    INFO(deriv_order_string + "Stationary spherical AH");
+    // Set all coefs back to zero first
+    initial_ah_coefs = zero_array;
+    initial_ah_coefs[0][iter.set(0, 0)()] = ah_radius;
+    expected_ah_coefs = {initial_time, initial_ah_coefs,
+                         std::numeric_limits<double>::infinity()};
+    test_shape_control(generator, make_not_null(&system_helper), initial_time,
+                       final_time, l_max, expected_ah_coefs, deriv_eps,
+                       deriv_eps);
+  }
+  system_helper.reset();
+  {
+    INFO(deriv_order_string + "Stationary non-spherical AH");
+    // Set all coefs back to zero first
+    initial_ah_coefs = zero_array;
+    for (iter.reset(); iter; ++iter) {
+      // Enforce l=0,l=1 components to be 0 always
+      if (iter.l() == 0 or iter.l() == 1) {
+        continue;
+      }
+      initial_ah_coefs[0][iter()] =
+          make_with_random_values<double>(generator, coef_dist, 1);
+    }
+    // Ensure that radius of AH is positive and constant
+    initial_ah_coefs[0][iter.set(0, 0)()] = ah_radius;
+    initial_ah_coefs[1] = zero_dv;
+    initial_ah_coefs[2] = zero_dv;
+    expected_ah_coefs = {initial_time, initial_ah_coefs,
+                         std::numeric_limits<double>::infinity()};
+    test_shape_control(generator, make_not_null(&system_helper), initial_time,
+                       final_time, l_max, expected_ah_coefs, deriv_eps,
+                       deriv_eps);
+    (void)func_eps;
+  }
+  system_helper.reset();
+  {
+    INFO(deriv_order_string + "AH coefficients linearly increasing/decreasing");
+    // Set all coefs back to zero first
+    initial_ah_coefs = zero_array;
+    for (iter.reset(); iter; ++iter) {
+      // Enforce l=0,l=1 components to be 0 always
+      if (iter.l() == 0 or iter.l() == 1) {
+        continue;
+      }
+      initial_ah_coefs[0][iter()] =
+          make_with_random_values<double>(generator, coef_dist, 1);
+      initial_ah_coefs[1][iter()] =
+          make_with_random_values<double>(generator, coef_dist, 1);
+    }
+    // Ensure that radius of AH is positive and constant
+    initial_ah_coefs[0][iter.set(0, 0)()] = ah_radius;
+    initial_ah_coefs[1][iter.set(0, 0)()] = 0.0;
+    initial_ah_coefs[2] = zero_dv;
+    expected_ah_coefs = {initial_time, initial_ah_coefs,
+                         std::numeric_limits<double>::infinity()};
+    test_shape_control(generator, make_not_null(&system_helper), initial_time,
+                       final_time, l_max, expected_ah_coefs, deriv_eps,
+                       deriv_eps);
+  }
+  system_helper.reset();
+  {
+    INFO(deriv_order_string +
+         "AH coefficients quadratically increasing/decreasing");
```

Comment:
I don't understand this case.  It looks like it will have a cubic part if DerivOrder is 3.  Also, why the division by `cube(i + 5)`?